apps/webservice/src/app/[workspaceSlug]/_components/DeleteDeployment.tsx

```diff
@@ -30,10 +30,21 @@ export const DeleteDeploymentDialog: React.FC<DeleteDeploymentProps> = ({
   const router = useRouter();
   const deleteDeployment = api.deployment.delete.useMutation();
 
+  const params = useParams<{
+    workspaceSlug: string;
+    systemSlug: string;
+  }>();
+
+  console.log("I am here DeleteDeployment");
+
   const onDelete = async () => {
-    await deleteDeployment.mutateAsync(id).catch(console.error);
-    router.refresh();
-    setIsOpen(false);
+    await deleteDeployment
+      .mutateAsync(id)
+      .then(() => {
+        router.push(`/${params.workspaceSlug}/systems/${params.systemSlug}`);
+        setIsOpen(false);
+      })
+      .catch(console.error);
```

Comment:
This ensures that if we delete from within the deployment view, we get redirected.

Reply:
need to router.refresh